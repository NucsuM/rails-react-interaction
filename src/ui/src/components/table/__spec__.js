import React from "react";
import Enzyme, { shallow, mount, configure } from "enzyme";
import Table from "./table";
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe("Table", () => {
  let wrapper;
  const sampleData = [
    {"id":1,
      "albumId":1,
      "title":"accusamus beatae ad facilis cum similique qui sunt",
      "url":"https://via.placeholder.com/600/92c952",
      "thumbnailUrl":"https://via.placeholder.com/150/92c952",
      "created_at":"2021-01-18T16:01:41.921Z",
      "updated_at":"2021-01-18T16:01:41.921Z"
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<Table tableContent={sampleData} />);
  });

  it("it displayes the content", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("it have the correct title", () => {
    expect(
      wrapper.contains(<td>accusamus beatae ad facilis cum similique qui sunt</td>)
    ).toBe(true);
    const tds = wrapper.find('td')
    expect(tds.at(2).props().children).toEqual('accusamus beatae ad facilis cum similique qui sunt')
  });
});
