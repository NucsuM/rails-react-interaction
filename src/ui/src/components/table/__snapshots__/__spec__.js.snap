// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table it displayes the content 1`] = `
<table>
  <thead>
    <tr>
      <th>
        ID
      </th>
      <th>
        albumId
      </th>
      <th>
        title
      </th>
      <th>
        url
      </th>
      <th>
        thumbnailUrl
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="1"
    >
      <td>
        1
      </td>
      <td>
        1
      </td>
      <td>
        accusamus beatae ad facilis cum similique qui sunt
      </td>
      <td>
        https://via.placeholder.com/600/92c952
      </td>
      <td>
        <img
          height="20"
          src="https://via.placeholder.com/150/92c952"
          width="20"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
